src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,129 @@
+import classNames from 'classnames';
+import {
+  ChangeEvent,
+  KeyboardEventHandler,
+  useCallback,
+  useState,
+} from 'react';
+
+import { TodoType } from '../../types/Todo';
+import { useTodos } from '../../contexts/todosContext';
+
+type TodoProps = {
+  todo: TodoType;
+  isProcessed: boolean;
+};
+
+export const Todo = ({ todo, isProcessed }: TodoProps) => {
+  const { completed, title } = todo;
+
+  const [editing, setEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(title);
+  const {
+    isLoading,
+    handleDeleteTodo,
+    handleToggleCompleted,
+    handleTitleUpdate,
+    addToProcessed,
+  } = useTodos();
+
+  const handleDeleteClick = useCallback(() => {
+    addToProcessed([todo]);
+    handleDeleteTodo(todo);
+  }, [todo]);
+
+  const onTodoTitleChange = useCallback(
+    (value: string) => {
+      if (!value) {
+        handleDeleteClick();
+
+        return;
+      }
+
+      if (title !== value) {
+        handleTitleUpdate(todo, value);
+      }
+
+      setEditing(false);
+    },
+    [todo],
+  );
+
+  const handleInputChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      e.preventDefault();
+
+      setInputValue(e.target.value);
+    },
+    [todo],
+  );
+
+  const handleKeyUp: KeyboardEventHandler<HTMLInputElement> = useCallback(
+    (e) => {
+      if (e.key === 'Escape') {
+        setEditing(false);
+        setInputValue(title);
+      }
+    },
+    [todo],
+  );
+
+  return (
+    <div
+      onDoubleClick={() => setEditing(true)}
+      className={classNames('todo', { completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          defaultChecked={completed}
```

Comment:
You should use the 'checked' property instead of 'defaultChecked' to make this input controlled.